CSS: additive-symbols descriptor (#29714)
mdn/content: files/en-us/web/css/@counter-style/additive-symbols/index.md

{{CSSRef}}

> The **`additive-symbols`** descriptor of the {{cssxref('@counter-style')}} at-rule is used to specify counter symbols when {{cssxref('@counter-style/system', 'system: additive')}} is set as a `@counter-style` descriptor. The additive system is used to construct [sign-value numbering](https://en.wikipedia.org/wiki/Sign-value_notation) systems such as Roman numerals.
> 
> The `additive-symbols` descriptor defines a comma-separated list of _additive tuples_. Each _additive tuple_ contains a space-separated non-negative integer weight and counter symbol. To be valid, the list must be in descending weight order.

Reviewer: - I find part of the previous phrasing better for flow and clarity
- We can move the second paragraph to "Description". There's also text after "Syntax" section, which too can be moved into "Description.
```suggestion
The **`additive-symbols`** descriptor of the {{cssxref('@counter-style')}} at-rule is used to specify counter symbols when the `@counter-style` {{cssxref('@counter-style/system', 'system')}} descriptor value is set as `additive`. The additive system is used to construct [sign-value numbering](https://en.wikipedia.org/wiki/Sign-value_notation) systems such as Roman numerals.
```

> ## Syntax

Reviewer: "Values" section is missing

```css

> additive-symbols: 3 "*";
> additive-symbols:
>   3 "0",

Reviewer: ```suggestion
/* Single tuple */
additive-symbols: 3 "*";

/* Comma-separated list of tuples */
additive-symbols:
  3 "0",
```

  2 "\2E\20",
  1 url(symbol.png);

> /* binary counter */

Reviewer: Added a blank line above comment
```suggestion

/* Binary counter */
```

additive-symbols:
  2 "1",
  1 "0";

> /* Etruscan counter */

Reviewer: Can we explain, perhaps in "Description", what binary and Etruscan counters are?
```suggestion

/* Etruscan counter */
```

additive-symbols:
  100 "𐌟",
  50 "𐌣",
  10 "𐌢",
  5 "𐌡",
  1 "𐌠";
```

> When the `system` descriptor is `cyclic`, `numeric`, `alphabetic`, `symbolic`, or `fixed`, use the {{cssxref('symbols')}} descriptor instead of `additive-symbols`.

Reviewer: - Moved the intro para here
- Do we need to say "weight"?
- Perhaps we can explain the "additive" part
```suggestion
## Description

The `additive-symbols` descriptor defines a comma-separated list of _additive tuples_. Each _additive tuple_ contains a space-separated non-negative integer and counter symbol. To be valid, the list must be in the descending order of integer. The integer and symbol are concatenated together to form the counter symbol.

When the `system` descriptor value is `cyclic`, `numeric`, `alphabetic`, `symbolic`, or `fixed`, use the {{cssxref('symbols')}} descriptor instead of `additive-symbols`.
```

{{CSSRef}}

The **`additive-symbols`** descriptor of the {{cssxref('@counter-style')}} at-rule is used to specify counter symbols when the `@counter-style` {{cssxref('@counter-style/system', 'system')}} descriptor value is set as `additive`. The additive system is used to construct [sign-value numbering](https://en.wikipedia.org/wiki/Sign-value_notation) systems such as Roman numerals.

## Syntax

```css
/* Single tuple */
additive-symbols: 3 "*";

/* Comma-separated list of tuples */
additive-symbols:
  3 "0",
  2 "\2E\20",
  1 url(symbol.png);

/* Binary counter */
additive-symbols:
  2 "1",
  1 "0";

/* Etruscan (a civilization in ancient Italy) counter  */
additive-symbols:
  100 "𐌟",
  50 "𐌣",
  10 "𐌢",
  5 "𐌡",
  1 "𐌠";
```

### Values

> The descriptor accepts a comma separated list of _additive tuples_ with each tuple consisting of the following two values separated by a space:

Reviewer: ```suggestion
The descriptor accepts a comma-separated list of _additive tuples_ with each tuple consisting of the following two values separated by a space:
```

> #### HTML

Reviewer: For a description of the example, we could say:

"In this example, `system: additive` along with the `additive-symbols` descriptor values specify how numbers should be represented as Roman numerals. The value of each `<li>` element in the list is converted to a Roman numeral according to the rules defined in `@counter-style`.

```html
<ul>
  <li>One</li>
  <li>Two</li>
  <li>Three</li>
  <li>Four</li>
  <li>Five</li>
  <li value="109">109</li>
  <li>110</li>
</ul>
```

#### CSS

```css
@counter-style uppercase-roman {
  system: additive;
  additive-symbols:
    1000 M,
    900 CM,
    500 D,
    400 CD,
    100 C,
    90 XC,
    50 L,
    40 XL,
    10 X,
    9 IX,
    5 V,
    4 IV,

>     1 I;

Reviewer: The height of the "Result" frame needs to be increased.

Reviewer: Also, can we add some text after the "Result" frame to explain the last two list items?

"For the list item with the `value` of `109`, the numeral `C` represents `100`, and `IX` represents `9`. This generates `CIX` counter for the list item `109`. The next list item automatically gets a `value` of `110`, and so the roman numeral is derived to be `CX`, `C` for `100` and `X` for `10`."

## See also

> - Other {{cssxref("@counter-style")}} descriptors, including, {{cssxref("@counter-style/system","system")}}, {{cssxref("@counter-style/symbols", "symbols")}}, {{cssxref("@counter-style/negative", "negative")}}, {{cssxref("@counter-style/prefix", "prefix")}}, {{cssxref("@counter-style/suffix", "suffix")}}, {{cssxref("@counter-style/range", "range")}}, {{cssxref("@counter-style/pad", "pad")}}, {{cssxref("@counter-style/speak-as", "speak-as")}}, and {{cssxref("@counter-style/fallback", "fallback")}}

Reviewer: ```suggestion
- {{cssxref("@counter-style")}} descriptors: {{cssxref("@counter-style/system","system")}}, {{cssxref("@counter-style/symbols", "symbols")}}, {{cssxref("@counter-style/negative", "negative")}}, {{cssxref("@counter-style/prefix", "prefix")}}, {{cssxref("@counter-style/suffix", "suffix")}}, {{cssxref("@counter-style/range", "range")}}, {{cssxref("@counter-style/pad", "pad")}}, {{cssxref("@counter-style/speak-as", "speak-as")}}, {{cssxref("@counter-style/fallback", "fallback")}}
```

> - {{Cssxref("list-style")}}, {{Cssxref("list-style-image")}}, {{Cssxref("list-style-position")}}
> - The {{cssxref("symbols", "symbols()")}}, functional notation is used for creating anonymous counter styles.
> - [CSS counter styles](/en-US/docs/Web/CSS/CSS_counter_styles) module

Reviewer: ```suggestion
- List style properties: {{Cssxref("list-style")}}, {{Cssxref("list-style-image")}}, {{Cssxref("list-style-position")}}
- {{cssxref("symbols", "symbols()")}} function to create anonymous counter styles
- [CSS counter styles](/en-US/docs/Web/CSS/CSS_counter_styles) module
```
